infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -946,6 +994,65 @@ private AddressMetaData writeRecord(SegmentHandle fh, long address,
         return new AddressMetaData(metadata.getChecksum(), metadata.getLength(), channelOffset);
     }
 
+    long getSegment(LogData entry) {
+        return entry.getGlobalAddress() / RECORDS_PER_LOG_FILE;
+    }
+
+    @Override
+    public void append(List<LogData> entries) {
+        // ignore empty list?
+
+        // check if the entries range cross a segment
+        LogData first = entries.get(0);
+        LogData last = entries.get(entries.size() - 1);
+        SegmentHandle firstSh = getSegmentHandleForAddress(first.getGlobalAddress());
```

Comment:
if first / last logData are holes, you will have -1 for global address. This could be very unfortunate. Maybe you should pass the range of addresses with the entries, so you don't have to rely on the getGlobalAddress. Another fix would be to have the address in a hole record. Not that we don't know it when we holefill...

Reply:
Good catch, I think this is a bug where the same Hole Entry can be shared for different address slots. Holes like every other entry has to have an address. I'll fix it in a different PR and make this one depend on it.

Reply:
Fixed.